### ISSUE TITLE
Can't deploy (from Cactus for Mac)

### ISSUE THREAD

**Issue by upsidaisium** [koenbok/Cactus#181]
I'm not entirely sure this is the right place to be posting issues related to Cactus for Mac, but I can't find anywhere else so here goes...

The deploy dialog says "Checking credentials..." as usual, but then then progress bar zips across and the green check mark icon displays as if deploy has finished successfully - but no files have actually been uploaded. There is a rather long traceback text in the error dialog that ends with...

    "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/ssl.py", line 271, in match_hostname % (hostname, ', '.join(map(repr, dnsnames)))) CertificateError: hostname u'ynnhoj.net.s3.amazonaws.com' doesn't match either of '*.s3.amazonaws.com', 's3.amazonaws.com'

I saw another issue on here that has been marked as resolved, which references something called boto, with a fix for this issue, but it doesn't make much sense to me. As far as I can tell boto is not installed on my computer. If it's included in the Cactus for Mac app then, well, I don't know how I can change any of the settings for it either.

Any help would be much appreciated.

**Comment by krallin** [comment]
This is indeed a duplicate of this other issue: https://github.com/koenbok/Cactus/issues/113, which is actually a boto bug (which I made a PR for, but it's never been accepted, no other fixes have been, to my knowledge). Have you tried the fix suggested there? Let me know if you need help doing that.

I don't know if changing your boto config as suggested there might fix your issue since you're using the Cactus app, but please try first.

I actually have a fix for this in a separate branch (https://github.com/koenbok/cactus/tree/ordinary-calling-format), but I'd have to merge that in first, and then @koenbok would need to make a new release.  

Cheers,

**Comment by upsidaisium** [comment]
Hi, and thanks for replying.

Before I posted this issue, I guess I didn't read the previous issue (#113) carefully enough, as I overlooked the suggestion to add...

    [s3]
    calling_format = boto.s3.connection.OrdinaryCallingFormat

...to `~/.boto`. I wasn't sure if the Cactus for Mac app would even look for config files in my home directory, because maybe the boto package had been modified before being inlcluded in Cactus for Mac, but I tried it anyway and deploying now works again. Phew! Thanks for pointing me in the right direction.
